components/tidb_query_expr/src/impl_encryption.rs

```diff
@@ -60,6 +64,78 @@ pub fn sha2(
     }
 }
 
+#[rpn_fn(nullable)]
+#[inline]
+pub fn compress(input: Option<BytesRef>) -> Result<Option<Bytes>> {
```

Comment:
Please considering use `writer` pattern here, take [this PR](https://github.com/tikv/tikv/pull/8728/files#diff-a06b8f2a39bb17e9932fc8d96118949a1f080cef810d7c35f3085b923f024fb2R18) as ref.

Reply:
done